dataset/src/test/scala/frameless/SelectTests.scala

```diff
@@ -94,4 +94,64 @@ class SelectTests extends TypedDatasetSuite {
 
     check(forAll(prop[Int, String, Double] _))
   }
+
+  test("select with obs +") {
+    def prop[A](data: Vector[X1[A]], const: A)(
+      implicit
+      eabc: TypedEncoder[X1[A]],
+      anum: Numeric[A],
+      eb: TypedEncoder[A]
+    ): Prop = {
+      val ds = TypedDataset.create(data)
+
+      val dataset2 = ds.select( ds('a) + const ).collect().run().toVector
+      val data2 = data.map { case X1(a) => anum.plus(a,const) }
+
+      dataset2 ?= data2
+    }
+
+    check(forAll(prop[Int] _))
+    check(forAll(prop[Long] _))
+    check(forAll(prop[Double] _))
+  }
+
+  test("select with obs *") {
```

Comment:
I had an issue with "/" because is not defined in "Numeric" is defined in "Fractional". I think if we move to our own types (where we can control everything and make sure they are all supported by Spark), then we can address this one as well.